src/nix/app.cc

```diff
@@ -121,8 +125,7 @@ UnresolvedApp InstallableValue::toApp(EvalState & state)
         }};
     }
 
-    else
```

Comment:
In case there are no matches with the previous cases, we always want to throw an error, so there's no need to use 'else'.